.github/workflows/release.yml

```diff
@@ -0,0 +1,38 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - master
```

Comment:
Does this mean that every time a PR is merged, or commits are pushed to master, there will be a new package release? I'm thinking perhaps we want to accumulate a few changes before each release?

Reply:
Whenever a push/PR merge happens to master, semantic release will check whether a new release is needed.
If it determines yes, then it will do the whole release process for us. If not, it won't do anything.

The way semantic release determines if a new release is needed is based of commit messages.
Using the default behavior, it will determine that a new release is needed if a commit message starts with "fix:", "feat:" or contains "BREAKING CHANGE:".
However in this PR, I've set it up to use [ESLint commit message linting](https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-eslint) so I believe it will make a release if a commit starts with "Fix:", "Update:", "New:" or "Breaking:".

I've been using semantic release in a few of my personal projects for a few years now and I've really been loving it. It has a huge amount of benefits over releasing manually.

Note: I've never set it up to use a non-default commit message style before so there might be a few teething issues - the eslint style doesn't seem to be a very popular style. It might end up being best to just use the default style.

Reply:
Ah, I see. So if we do commits with eg. "Build:", "Docs:" and similar prefixes in the eslint standard those will not trigger a release.

How about a scenario where we have two PRs, both prefixed with "Breaking:", and we merge them in the same day. Will merging them increment the major version two steps?

Reply:
If they are merged one after the other with a largish time span in between (like >1 minute (however long it takes semantic release to do its thing)), then yes.
If they are merged one straight after the other then no, as semantic release will only do a release if what it is looking at is the current head on the upstream. So essentially the second one cancels the first release so only one release happens.

Theoretically we could add an arbitrary delay between when the push happens and when semantic release runs so we wouldn't have to race it.

However that being said, I think the best approach to take if this situation arises would be to merge the two branches into a new 3rd branch and then merge that branch into master. That 3rd branch would probably have a name like "next" as it's plan to be the next major release. It could also sit around for awhile an accumulate lots of other changes until we are to release them all at once.

I like to think of it from the mindset of merging into master is essentially saying "release this thing now".

Reply:
I've been thinking about this a bit and I think it's best to use the more traditional commit style (angular style). With customizations to back it cool 😎